includes/reports/class-sensei-no-users-table-relationship.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * File containing the Sensei_No_Users_Table_Relationship class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class that handles a special case of environments that have the users table in a separate
+ * database from the other tables.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_No_Users_Table_Relationship {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_No_Users_Table_Relationship constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
```

Comment:
Maybe another type hint?

```suggestion
	public function init() : void {
```

Reply:
Thinking a bit more about this, do we need this method? I saw that most classes registered on `Sensei_Main::initialize_global_objects` just register the actions directly on the constructor... :thinking: